plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1156,7 +1156,7 @@ public VBD createVbd(final Connection conn, final DiskTO volume, final String vm
         if (volume.getType() == Volume.Type.ISO) {
             vbdr.mode = Types.VbdMode.RO;
             vbdr.type = Types.VbdType.CD;
-            vbdr.userdevice = "3";
+            vbdr.userdevice = String.valueOf(3 + isoCount);
```

Comment:
maybe give the 3 a constant with descriptive name

Reply:
@DaanHoogland I have made the changes.